### DISCUSSION TITLE
Aqara and Xiaomi Firmware grabbing without having the physical devices

### DISCUSSION THREAD

**Discussion by sorryusernameisalreadytaken** [Koenkk/zigbee2mqtt#13193] [Devices]
I see here that the user @Otnow was able to download aqara firmwares without having the devices.

How is that possible? I know it could be a secret but having the latest fw's always would be nice. Why is there no chance to run a kind of checking service to push always the latest fw into z2m?

**Comment by Otnow** [comment]
Yes, I was recently able to understand how you can get links to OTA files for _Xiaomi/Aqara_ without having physical devices.

I need more time to try to get additional benefit from this method and after that I plan to publish the results of my research on this topic.

**Reply by Dunkelschunkel** [reply]
Wow, that would be a banger.
Would this work for tuya devices as well?

**Reply by Otnow** [reply]
> Would this work for tuya devices as well?

Unfortunately not, only for _Xiaomi/Aqara_.

**Reply by Dunkelschunkel** [reply]
Thank you for clarifying

**Reply by henkiejan1** [reply]
Should be nice also for the Xiaomi / Aqara temperature, humidity and pressure sensor WSDCGQ11LM. Current firmware is from 05-12-2019 so a new firmware could be bring fixes in the sometime troubles with refreshes i think.

**Reply by sorryusernameisalreadytaken** [reply]
@Otnow I don't think you need to place the results here. If you are worried that the "gap" will be closed by Xiaomi after the release, let's design a docker or something similar that runs on a server or raspberry-pi at home. With 3 or 4 people on private registy. This can then load images and firmwares from the servers and then provides the links in pr's for z2m, deconz and so on.

**Reply by Otnow** [reply]
> Should be nice also for the Xiaomi / Aqara temperature, humidity and pressure sensor WSDCGQ11LM. Current firmware is from 05-12-2019 so a new firmware could be bring fixes in the sometime troubles with refreshes i think.

Unfortunately, it is not possible to update the firmware for _WSDCGQ11LM_, as it does not have an OTA function (no OTA cluster).

**Reply by Otnow** [reply]
> @Otnow I don't think you need to place the results here. If you are worried that the "gap" will be closed by Xiaomi after the release, let's design a docker or something similar that runs on a server or raspberry-pi at home. With 3 or 4 people on private registy. This can then load images and firmwares from the servers and then provides the links in pr's for z2m, deconz and so on.

In addition to getting links to firmware, the found method provides additional features, so I will not hide it.
The work has slowed down but is still going on.

**Comment by mclever** [comment]
@Otnow did you post somewhere else or is this still the thread to watch? Thanks

**Reply by Otnow** [reply]
Work is still in progress, so the result has not yet been published anywhere.

**Reply by z-x-f** [reply]
> Work is still in progress, so the result has not yet been published anywhere.

Hi, what is the progress so far?

**Reply by henkiejan1** [reply]
> Work is still in progress, so the result has not yet been published anywhere.

Any updates?
